osu.Game.Rulesets.Osu/Difficulty/Evaluators/AimEvaluator.cs

```diff
@@ -112,18 +118,50 @@ public static double EvaluateDifficultyOf(DifficultyHitObject current, bool with
                 velocityChangeBonus *= Math.Pow(Math.Min(osuCurrObj.StrainTime, osuLastObj.StrainTime) / Math.Max(osuCurrObj.StrainTime, osuLastObj.StrainTime), 2);
             }
 
-            if (osuLastObj.BaseObject is Slider)
+            // Add in acute angle bonus or wide angle bonus + velocity change bonus, whichever is larger.
+            aimStrain += Math.Max(acuteAngleBonus * acute_angle_multiplier, wideAngleBonus * wide_angle_multiplier + velocityChangeBonus * velocity_change_multiplier);
+
+            double sliderJumpBonus = 0;
             {
-                // Reward sliders based on velocity.
-                sliderBonus = osuLastObj.TravelDistance / osuLastObj.TravelTime;
+                if (osuLastObj.BaseObject is Slider slider)
+                {
+                    // Take just slider heads into account because we're computing sliderjumps, not slideraim
+                    sliderJumpBonus = slider_jump_multiplier * aimStrain * sliderJumpsAdjustingMultiplier;
+
+                    // Reward more if sliders and circles are alternating (actually it's still lower than several sliders in a row)
+                    if (osuLastLastObj.BaseObject is HitCircle)
+                    {
+                        double alternatingBonus = 0.5 * slider_jump_multiplier * osuLastObj.JumpDistance / osuLastObj.StrainTime;
+
+                        if (osuLastObj.StrainTime > osuLastLastObj.StrainTime)
+                            alternatingBonus *= Math.Pow(Math.Min(osuCurrObj.StrainTime, osuLastObj.StrainTime) / Math.Max(osuCurrObj.StrainTime, osuLastObj.StrainTime), 2);
+
+                        sliderJumpBonus += alternatingBonus;
+                    }
+
+                    // If slider was slower than notes before - punish it
+                    if (osuCurrObj.StrainTime > osuLastObj.StrainTime)
+                        sliderJumpBonus *= Math.Pow(Math.Min(osuCurrObj.StrainTime, osuLastObj.StrainTime) / Math.Max(osuCurrObj.StrainTime, osuLastObj.StrainTime), 2);
+
+                    // Punish too short sliders to prevent cheesing (cheesing is still possible, but it's very rare)
+                    double sliderLength = slider.Velocity * slider.SpanDuration;
```

Comment:
There's an inconsistency here, in the speed evaluator you are taking `sliderLength` as `2 * slider.Velocity * slider.SpanDuration`. I assume this isn't intended, and if it is then it's weird.

Reply:
i just want to punish short sliders in speed 2 times less than in aim
but i will fix the fact that it's looking confusing